src/runtime/virtcontainers/clh.go

```diff
@@ -395,8 +425,33 @@ func (clh *cloudHypervisor) CreateVM(ctx context.Context, id string, network Net
 
 	// Create the VM memory config via the constructor to ensure default values are properly assigned
 	clh.vmconfig.Memory = chclient.NewMemoryConfig(int64((utils.MemUnit(clh.config.MemorySize) * utils.MiB).ToBytes()))
-	// shared memory should be enabled if using vhost-user(kata uses virtiofsd)
-	clh.vmconfig.Memory.Shared = func(b bool) *bool { return &b }(true)
+
+	if !clh.config.BootFromTemplate {
+		// shared memory should be enabled if using vhost-user(kata uses virtiofsd)
+		// or VM templating builds the first VM with file-backed memory
+		clh.vmconfig.Memory.Shared = func(b bool) *bool { return &b }(true)
+	}
```

Comment:
I wonder why we need to explicitly call out the case for `!clh.config.BootFromTemplate` and not enable shared memory?

I think the code would be cleaner to always enable `clh.vmconfig.Memory.Shared` (no matter it is vm-template or not), while ensure `memoryZoneConfig.Shared` is also enabled for the vm-template case (both `clh.config.BootToBeTemplate` and `clh.config.BootFromTemplate`).

Reply:
Thanks for your review, you are right, i will delete it.

Reply:
Thanks for your review, already been deleted.